charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -243,11 +243,7 @@ spec:
           name: consul-data
           readOnly: false
         {{- if .Values.global.tls.enabled }}
-        {{- if .Values.global.tls.enableAutoEncrypt }}
-        - name: consul-auto-encrypt-ca-cert
-        {{- else }}
         - name: consul-ca-cert
-        {{- end }}
           mountPath: /consul/tls/ca
           readOnly: true
         {{- end }}
```

Comment:
nit: `-datacenter`, `-partition` and `-api-timeout` flag values below are duplicated because they're already provided as env vars `{{- include "consul.consulK8sConsulServerEnvVars" . | nindent 8 }}` I believe. It'd be nice if we removed them to avoid any future weirdness.

Reply:
Great! I did this apart from the `-datacenter` flag. I noticed that `acl-init` is using Consul datacenter instead of Consul login datacenter and if this is removed, the acl init will use the local dc. I will follow up on whether we need to change `acl-init` under the hood to use Consul login datacenter instead.